src/freemindlatex/compilation_server.py

```diff
@@ -0,0 +1,292 @@
+import codecs
+import collections
+import errno
+import logging
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+
+from freemindlatex import compilation_service, convert
+from jsonrpc import JSONRPCResponseManager, dispatcher
+from werkzeug.serving import run_simple
+from werkzeug.wrappers import Request, Response
+
+
+def _MkdirP(directory):
+  """Makes sure the directory exists. Otherwise, will try creating it.
+
+  Args:
+    directory: the directory to make sure exist
+
+  Returns:
+    Nothing
+
+  Raises:
+    OSError, when encounter problems creating the directory.
+  """
+
+  try:
+    os.makedirs(directory)
+  except OSError as exc:  # Python >2.5
+    if exc.errno == errno.EEXIST and os.path.isdir(directory):
+      pass
+    else:
+      raise
+
+
+def _CompileLatexAtDir(working_dir, filename,
```

Comment:
the "filename" argument is confusing, and unnecessary. It should always be "slides.tex". Make it a flag.